### DISCUSSION TITLE
Change 'max_len' during learning

### DISCUSSION THREAD

**Discussion by junylee11** [yl4579/StyleTTS2#266] [Q&A]
I'm doing stage 1 learning up to 20epoch, and stage 2 learning. I set the value of 'max_len' to 800, which takes a very long time to learn. (Over 20 hours per epoch)
Therefore, I want to reduce the max_len value from 800 to 400.

1. Would this reduce the speed of learning be doubled?
2. Will it have an adverse impact on quality? ("Note that finetune"says that changing max_len during learning has an adverse impact, but I wonder how much impact it has.)

**Comment by martinambrus** [comment]
Lowering `max_len` will always have an adverse impact on the output quality, since it defines how much of your audio is being processed from each file. For example, your 800 current `max_len` will process up to 10 seconds of audio from each WAV file ( 800 x 0.0125 = 10 ). If you lower this value, the training will only use 5 seconds from each of the audio file instead, thus lowering the quality by potentially cutting a word / sentence mid-way.

I would recommend always setting `max_len` to the length that fits your longest audio file. Or - if you need to lower that value - remove all audio files that are longer than the new `max_len` value, so you'll make sure all your WAV files get processed in their fullness. This will potentially still lower the output quality (since the model will only ever learn from short sentences and may struggle with longer ones) but at least it'll learn how to infer those sentences correctly, without any strange noises or artifacts in them.
